onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -104,6 +104,10 @@ TensorrtExecutionProvider::TensorrtExecutionProvider(const TensorrtExecutionProv
   const char* workspace_env = getenv("ORT_TENSORRT_MAX_WORKSPACE_SIZE");
   if (workspace_env)
     max_workspace_size_ = atoi(workspace_env);
+
+  const char* fp16_env = getenv("ORT_TENSORRT_FP16_ENABLE");
```

Comment:
for all these getenv calls, 
on windows it's better to use _dupenv_s  (which is more secure)
see https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/ngraph/ngraph_custom_op.cc#L31 
as example.

Reply:
I suggest we use the GetEnvironmentVar function in the Env class that abstracts out the specific platform. Better than duplicating this in multiple EPs.